WORKSPACE

```diff
@@ -205,3 +205,173 @@ protoc_setup()
 # Loads all of the C dependencies that we rely on.
 load("//misc/bazel/c_deps:repositories.bzl", "c_repositories")
 c_repositories()
+
+# `rules_rust`
+#
+# Rules for building Rust crates, and several convienence macros for building all transitive
+# dependencies.
+
+RULES_RUST_VERSION = "0.44.0"
+RULES_RUST_INTEGRITY = "sha256-pt9MIrs/tDVzpMQIjvbQ+v44oOzQ+FrSH/2IiAAzcDA="
+
+maybe(
+    http_archive,
+    name = "rules_rust",
+    integrity = RULES_RUST_INTEGRITY,
+    urls = [
+        "https://github.com/bazelbuild/rules_rust/releases/download/{0}/rules_rust-v{0}.tar.gz".format(RULES_RUST_VERSION),
+        "https://mirror.bazel.build/bazelbuild/rules_rust/releases/download/{0}/rules_rust-v{0}.tar.gz".format(RULES_RUST_VERSION),
+    ],
+)
+
+load("@rules_rust//rust:repositories.bzl", "rules_rust_dependencies")
+rules_rust_dependencies()
+
+# `rustc`
+#
+# Fetch and register the relevant Rust toolchains. We use a custom macro that
+# depends on `rules_rust` but cuts down on bloat from their defaults.
+
+RUST_VERSION = "1.78.0"
+
+load("//misc/bazel/toolchains:rust.bzl", "rust_toolchains")
+rust_toolchains(
+    RUST_VERSION,
+    [
+        "aarch64-apple-darwin",
+        "aarch64-unknown-linux-gnu",
+        "x86_64-unknown-linux-gnu",
+        "x86_64-apple-darwin",
```

Comment:
Are there any users of this?

Reply:
Not yet, but we do cross compiling today in CI so I want to make sure that stays supported in Bazel